addons/google_drive/models/google_drive.py

```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+import re
+import urllib2
+import werkzeug.urls
+from openerp import api, fields, models, _
+from openerp.exceptions import RedirectWarning, UserError
+from openerp.tools.safe_eval import safe_eval as eval
+
+from openerp.addons.google_account import TIMEOUT
+
+
+class GoogleDriveConfig(models.Model):
+    _name = 'google.drive.config'
+    _description = "Google Drive templates config"
+
+    name = fields.Char('Template Name', required=True)
+    model_id = fields.Many2one('ir.model', 'Model', ondelete='set null', required=True)
+    model = fields.Char(related='model_id.model', readonly=True)
+    filter_id = fields.Many2one('ir.filters', 'Filter', domain="[('model_id', '=', model)]")
+    google_drive_template_url = fields.Char('Template URL', required=True)
+    google_drive_resource_id = fields.Char(compute='_compute_resource_id', string='Resource ID')
+    google_drive_client_id = fields.Char(compute='_compute_client_id', string='Google Client')
+    name_template = fields.Char('Google Drive Name Pattern', required=True, default="Document %(name)s",
+                                help='Choose how the new google drive will be named, on google side. Eg. gdoc_%(field_name)s')
+    active = fields.Boolean('Active', default=True)
+
+    @api.multi
+    def _compute_resource_id(self):
+        for config in self:
+            key = config._get_key_from_url(config.google_drive_template_url)
+            if key:
+                config.google_drive_resource_id = key
+            else:
+                raise UserError(_("Please enter a valid Google Document URL."))
+
+    @api.multi
+    def _compute_client_id(self):
+        client_id = self.env['ir.config_parameter'].get_param('google_drive_client_id')
+        self.write({'google_drive_client_id': client_id})
+
+    @api.one
+    @api.constrains('model_id', 'filter_id')
+    def _check_model_id(self):
+        if self.filter_id and self.model_id.model != self.filter_id.model_id:
+            raise UserError(_('Model of selected filter is not matching with model of current template.'))
+        return True
+
+    @api.onchange('model_id')
+    def onchange_model_id(self):
+        if self.model_id:
+            self.model = self.model_id.model
+        else:
```

Comment:
may be no else required do like this way: 
if not self.model_id: 
      self.model = False
      self.filter_id = False
self.model = self.model_id.model